File: src/main/java/me/senseiwells/arucas/api/docs/ClassDoc.java
package me.senseiwells.arucas.api.docs;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface ClassDoc {
	String name();
	String[] desc();
	String importPath() default "";
}

File: src/main/kotlin/me/senseiwells/arucas/api/docs/ClassDoc.kt
package me.senseiwells.arucas.api.docs

import me.senseiwells.arucas.builtin.ObjectDef
import me.senseiwells.arucas.utils.Util
import kotlin.reflect.KClass

@Deprecated("me.senseiwells.arucas.api.docs.annotations.ClassDoc should be used instead.")
@Target(AnnotationTarget.CLASS)
annotation class ClassDoc(
    val name: String,
    val desc: Array<String>,
    val importPath: String = "",
    val superclass: KClass<*> = ObjectDef::class,
    val language: Util.Language = Util.Language.Kotlin
)
